### ISSUE TITLE
Cant toggle local video track

### ISSUE THREAD

**Issue by robschwitzer** [twilio/twilio-video.js#1010]
Hey folks,

Have been encountering an issue with toggling local media. Turning it off works as expected, but turning it back on doesn't add the video back to the screen. The track is successfully created and the camera turns on, but no video. Any suggestions would be appreciated, thanks!

 - [ ✔️] I have verified that the issue occurs with the latest twilio-video.js release and is not marked as a known issue in the [CHANGELOG.md](https://github.com/twilio/twilio-video.js/blob/master/CHANGELOG.md).
 - [ ✔️] I reviewed the [Common Issues](https://github.com/twilio/twilio-video.js/blob/master/COMMON_ISSUES.md) and open GitHub issues and verified that this report represents a potentially new issue.
 - [✔️ ] I verified that the [Quickstart](https://github.com/twilio/video-quickstart-js) application works in my environment.
 - [ ✔️] I am not sharing any [Personally Identifiable Information (PII)](https://www.twilio.com/docs/glossary/what-is-personally-identifiable-information-pii)
       or sensitive account information (API keys, credentials, etc.) when reporting this issue.

**Code to reproduce the issue:**

```
// add twilio
<script src="//media.twiliocdn.com/sdk/js/video/releases/2.4.0/twilio-video.min.js"></script>

// connect to twilio
      twilio
        .connect(token, {
          name: roomName,
          audio: true,
          video: { height: 720, frameRate: 24, width: 1280 },
          bandwidthProfile: {
            video: {
              mode: "collaboration",
              maxTracks: 2,
              dominantSpeakerPriority: "standard",
              renderDimensions: {
                high: { height: 1080, width: 1980 },
                standard: { height: 720, width: 1280 },
                low: { height: 176, width: 144 },
              },
            },
          },
          dominantSpeaker: true,
          maxAudioBitrate: 16000,
          preferredVideoCodecs: [{ codec: "VP8", simulcast: true }],
          networkQuality: { local: 1, remote: 1 },
        })
        .then(room => {
          setRoom(room);
          room.on("participantConnected", participantConnected);
          room.on("participantDisconnected", participantDisconnected);
          room.participants.forEach(participantConnected);
        })

  const participantConnected = participant => {
    setParticipants(prevParticipants => [...prevParticipants, participant]);
  };

  const participantDisconnected = participant => {
    setParticipants(prevParticipants =>
      prevParticipants.filter(p => p !== participant)
    );
  };

// toggle video

// participant == room.localParticipant
// videoRef is a ref to #local-participant

  const toggleLocalVideo = () => {
    if (isVideoEnabled) {
      const track = [...participant.videoTracks.values()][0].track;
      track.stop();
      track.detach().forEach(element => element.remove());
      participant.unpublishTrack(track);
    } else {
      twilio.createLocalVideoTrack().then((track) => {
        // nothing happens 
        // const localVideo = document.querySelector(`#local-participant`);
        // localVideo.appendChild(track.attach());
        track.attach(videoRef.current);
        return participant.publishTrack(track);
      });
    }
    setVideoEnabled(!isVideoEnabled);
  };
```

**Expected behavior:**

When the video is ON, and i want to turn it OFF, the local video should disappear and the camera light should turn off.
When the video is OFF, and i want to turn it ON, the local video should re-appear 

**Actual behavior:**

When the video is ON, and i want to turn it OFF, the local video disappears and the camera light turns off.
When the video is OFF, and i want to turn it ON, the camera light turns on but the video does not re-appear.

**LOGS:**

Turn video off:
```
INFO in [LocalVideoTrack #2: 381d78c5-aa46-4c66-bfe6-72b7450a010d]: Stopping
DEBUG in [LocalVideoTrack #2: 381d78c5-aa46-4c66-bfe6-72b7450a010d]: Ended
DEBUG in [LocalVideoTrack #2: 381d78c5-aa46-4c66-bfe6-72b7450a010d]: Attempting to detach from elements: [video#local-participant.sc-fzXfMy.bcpgoE]
INFO in [LocalParticipant #1: PA21cce9dcb0e894ec17ffe01787d4f0f5]: Removed a VideoTrack: 381d78c5-aa46-4c66-bfe6-72b7450a010d
DEBUG in [LocalParticipant #1: PA21cce9dcb0e894ec17ffe01787d4f0f5]: VideoTrack: LocalVideoTrack {kind: "video", name: "381d78c5-aa46-4c66-bfe6-72b7450a010d", mediaStreamTrack: MediaStreamTrack, dimensions: {…}, …}
INFO in [LocalParticipant #1: PA21cce9dcb0e894ec17ffe01787d4f0f5]: Removed a LocalVideoTrack: 381d78c5-aa46-4c66-bfe6-72b7450a010d
DEBUG in [LocalParticipant #1: PA21cce9dcb0e894ec17ffe01787d4f0f5]: LocalVideoTrack: LocalVideoTrack {kind: "video", name: "381d78c5-aa46-4c66-bfe6-72b7450a010d", mediaStreamTrack: MediaStreamTrack, dimensions: {…}, …}
INFO in [LocalParticipant #1: PA21cce9dcb0e894ec17ffe01787d4f0f5]: Removed a VideoTrackPublication: MT0dcf541848b94933a70f7def9f3a6612
DEBUG in [LocalParticipant #1: PA21cce9dcb0e894ec17ffe01787d4f0f5]: VideoTrackPublication: LocalVideoTrackPublication {trackName: "381d78c5-aa46-4c66-bfe6-72b7450a010d", trackSid: "MT0dcf541848b94933a70f7def9f3a6612", kind: "video", …}
```

Turn video on:
```
INFO in [LocalParticipant #1: PA21cce9dcb0e894ec17ffe01787d4f0f5]: Added a new VideoTrack: 13238d7d-0809-45d6-90e9-371c222c7812
DEBUG in [LocalParticipant #1: PA21cce9dcb0e894ec17ffe01787d4f0f5]: VideoTrack: LocalVideoTrack {kind: "video", name: "13238d7d-0809-45d6-90e9-371c222c7812", mediaStreamTrack: MediaStreamTrack, dimensions: {…}, …}
INFO in [LocalParticipant #1: PA21cce9dcb0e894ec17ffe01787d4f0f5]: Added a new LocalVideoTrack: 13238d7d-0809-45d6-90e9-371c222c7812
DEBUG in [LocalParticipant #1: PA21cce9dcb0e894ec17ffe01787d4f0f5]: LocalVideoTrack: LocalVideoTrack {kind: "video", name: "13238d7d-0809-45d6-90e9-371c222c7812", mediaStreamTrack: MediaStreamTrack, dimensions: {…}, …}
```

**Software versions:**

 - [ ✔️] Browser(s): Chrome Version 81.0.4044.122
 - [ ✔️] Operating System: MacOS 10.15.4
 - [ ✔️] twilio-video.js: 2.4.0
 - [ ✔️] Third-party libraries (e.g., Angular, React, etc.): React 16.12.0

**Comment by manjeshbhargav** [comment]
Hi @robschwitzer ,

When you are detaching the old VideoTrack, you are also removing the video element from the DOM. When you are attaching the new VideoTrack, `videoRef.current` no longer exists in the DOM. So please detach the old VideoTrack without removing it from the DOM as shown below:
```js
const toggleLocalVideo = () => {
    if (isVideoEnabled) {
      const track = [...participant.videoTracks.values()][0].track;
      track.stop();
      track.detach(videoRef.current);
      participant.unpublishTrack(track);
    } else {
      twilio.createLocalVideoTrack().then(track => {
        track.attach(videoRef.current);
        return participant.publishTrack(track);
      });
    }
    setVideoEnabled(!isVideoEnabled);
  };
```
Please let me know if this works for you.

Thanks,

Manjesh Malavalli
JSDK Team

**Comment by robschwitzer** [comment]
thanks @manjeshbhargav , that makes sense now
